src/native/http_request_utils.c

```diff
@@ -177,10 +178,86 @@ struct aws_input_stream *aws_input_stream_new_from_java_http_request_body_stream
     return NULL;
 }
 
+static inline int s_marshal_http_header_to_buffer(
+    struct aws_byte_buf *buf,
+    const struct aws_byte_cursor *name,
+    const struct aws_byte_cursor *value) {
+    if (aws_byte_buf_reserve_relative(buf, sizeof(int) + sizeof(int) + name->len + value->len)) {
+        return AWS_OP_ERR;
+    }
+
+    aws_byte_buf_write_be32(buf, (uint32_t)name->len);
```

Comment:
Dumb q, but do we know that Java's deserialize/serialize are big endian?

Reply:
yes. Java is always big endian. Unfortunately java doesn't provide a mechanism to check host ordering easily on the java side of things, so we just always do BE on the jni side.

Reply:
plus we control the code on both sides of the boundary